core/src/main/scala/pink/cozydev/lucille/Parser.scala

```diff
@@ -115,6 +115,23 @@ object Parser {
   def unaryMinus(query: P[Query]): P[UnaryMinus] =
     P.char('-') *> query.map(UnaryMinus.apply)
 
+  // Range query
+  // e.g. '{cats TO dogs}', '[1 TO *}'
+  def rangeQuery: P[RangeQ] = {
+    val inclLower = P.charIn('{', '[').map(lowerBound => lowerBound == '[') <* maybeSpace
+    val inclUpper = maybeSpace *> P.charIn('}', ']').map(upperBound => upperBound == ']')
+    val boundValue =
+      P.char('*').as(None) | P.not(P.stringIn(reserved)).with1 *> (alpha | digit | P.char(
```

Comment:
We had some discussion about whether or not we should allow `{cats TO NOT}`, and decided to leave this as is for now and check with Lucene.
I kind of suspect it's allowed as a negation doesn't make sense in that position. But we should check to be sure.